src/cryptsetup/cryptsetup.c

```diff
@@ -1115,9 +1116,10 @@ static int attach_tcrypt(
         if (arg_tcrypt_veracrypt && arg_tcrypt_veracrypt_pim != 0)
                 params.veracrypt_pim = arg_tcrypt_veracrypt_pim;
 
-        if (key_data) {
-                params.passphrase = key_data;
-                params.passphrase_size = key_data_size;
+        /* We don't use iovec_is_set here to allow empty passwords (iov_base != NULL && iov_len == 0) */
+        if (key_data->iov_base) {
```

Comment:
hmm, I think we should instead check for `key_data` itsefl being non-NULL.

Reply:
But then we would need to do `key_data && key_data->iov_base`, no? Because we could receive a zero-initialized iovec.

Reply:
Oh, nevermind I see what you meant, done.